From: Kate Hsuan <hpa@redhat.com>
Date: Wed, 23 Jul 2025 19:14:01 +0800
Subject: gio: gmemorymonitorpoll: Fix unit test error since the callback
 returns earlier

The value overriding has to be set before testing the value and returning
the callback. Otherwise, the callback can't emit a signal to the test
program.
---
 gio/gmemorymonitorpoll.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gio/gmemorymonitorpoll.c b/gio/gmemorymonitorpoll.c
index d5d4e75..55ca546 100644
--- a/gio/gmemorymonitorpoll.c
+++ b/gio/gmemorymonitorpoll.c
@@ -150,16 +150,16 @@ g_memory_monitor_mem_ratio_cb (gpointer data)
 
   mem_ratio = g_memory_monitor_base_query_mem_ratio ();
 
+  /* free ratio override */
+  if (monitor->mem_free_ratio >= 0.0)
+    mem_ratio = monitor->mem_free_ratio;
+
   if (mem_ratio < 0.0)
     return G_SOURCE_REMOVE;
 
   if (mem_ratio > 0.5)
     return G_SOURCE_CONTINUE;
 
-  /* free ratio override */
-  if (monitor->mem_free_ratio >= 0.0)
-    mem_ratio = monitor->mem_free_ratio;
-
   g_debug ("memory free ratio %f", mem_ratio);
 
   if (mem_ratio < 0.2)
